src/main/java/echoquery/intents/NarrowHandler.java

```diff
@@ -0,0 +1,84 @@
+package echoquery.intents;
+
+import java.io.IOException;
+import java.sql.Connection;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazon.speech.slu.Intent;
+import com.amazon.speech.speechlet.Session;
+import com.amazon.speech.speechlet.SpeechletResponse;
+
+import echoquery.sql.Querier;
+import echoquery.sql.QueryRequest;
+import echoquery.sql.QueryResult;
+import echoquery.sql.model.ColumnType;
+import echoquery.utils.Response;
+import echoquery.utils.Serializer;
+import echoquery.utils.SessionUtil;
+import echoquery.utils.SlotUtil;
+
+public class NarrowHandler implements IntentHandler {
+
+  private static final Logger log =
+      LoggerFactory.getLogger(NarrowHandler.class);
+  private final Querier querier;
+
+  public NarrowHandler(Connection conn) {
+    this.querier = new Querier(conn);
+  }
+
+  @Override
+  public SpeechletResponse respond(Intent intent, Session session) {
+    try {
+      QueryRequest request = 
+          updateRequest(intent, (QueryRequest) Serializer.deserialize(
+              (String) session.getAttribute(SessionUtil.REQUEST_ATTRIBUTE)));
+      session.setAttribute(
```

Comment:
and the rest of this method, have the constructor take in an AggregationHandler and do,

aggregationHandler.respond(updateRequest(intent, request), session)

That should take care of everything and make sure that if we change something in the future we'll only have to do it in aggregationHandler